src/elitefurretai/utils/team_repo.py

```diff
@@ -13,14 +14,64 @@ class TeamRepo:
 
     def __init__(self, filepath: str = "data/teams"):
         self._teams: Dict[str, str] = {}
-        files = [
-            os.path.join(filepath, f)
-            for f in os.listdir(filepath)
-            if os.path.isfile(os.path.join(filepath, f))
-        ]
-        for file in files:
-            with open(file, "r") as f:
-                self._teams[os.path.basename(file).replace(".txt", "")] = f.read()
+
+        for format_folder in os.listdir(filepath):
+            format_name = format_folder
+            folder_path = os.path.join(filepath, format_folder)
+            if os.path.isdir(folder_path):
+                print(f"format name: {format_name}")
+                filenames = os.listdir(folder_path)
+                for filename in filenames:
+                    team_path = os.path.join(filepath, format_folder, filename)
+                    print(team_path)
+                    with open(team_path, "r") as f:
+                        team_string = f.read()
+                    self._teams[
+                        format_name + "_" + os.path.basename(team_path).replace(".txt", "")
+                    ] = team_string
+                    self.validate_team(team_string, format_name)
+
+    # Returns True if valid
+    def validate_team(
```

Comment:
This looks great -- but want to confirm you've tested this by calling it in multiple directories?

Reply:
I didn't! :scream:
(Just tested now and it is causing a problem, will fix)
Good catch